### ISSUE TITLE
Syntax highlighting in blame mode broken (or different from normal mode) for some files

### ISSUE THREAD

**Issue by eric-j-ason** [go-gitea/gitea#24383]
### Description

Some files get highlighted differently when viewed in blame mode as compared to when viewed in normal mode. [Example](https://try.gitea.io/baah/terminal/src/commit/f7a6c43aeee0cf9e188b6ea4eae3a4b8c02d8718/XamlStyler.json).

Most files beginning with `#!/usr/bin/env bash` do not get highlighted at all in blame mode. [Example](https://codeberg.org/blau_araujo/vrms-bash/blame/commit/49f506623f8d681f9912747dc1adf750d8ea49b6/vrms). I found [one example](https://codeberg.org/diva.exchange/divachain/blame/commit/0f0104365d20a5571a031ba334d73435149100f2/bin/build.sh) of `#!/usr/bin/env bash` that _does_ get highlighted, but when I put the same file on my own server, the highlighting breaks also for that file.

### Gitea Version

1.20.0+dev-566-g8cc60ee01

### Can you reproduce the bug on the Gitea demo site?

Yes

### Screenshots

![bild](https://user-images.githubusercontent.com/75301977/234867286-0bdab3a6-ab32-4cb9-beec-aa1a06e93f7d.png)
![bild](https://user-images.githubusercontent.com/75301977/234867336-9bb8b31d-c273-44fb-944a-a9e1507ed863.png)

### Git Version

Whatever is on <try.gitea.io>

### Operating System

Whatever is on <try.gitea.io>

### How are you running Gitea?

try.gitea.io

### Database

None

**Comment by silverwind** [comment]
> Dup of https://github.com/go-gitea/gitea/issues/23176 ?

I think not, because in the blame case, the highlighter has all all context so should produce the same highlighting as the file view.

**Comment by wxiaoguang** [comment]
I guess "Blame" also renders the content block by block, no full content.

**Comment by silverwind** [comment]
Yes, that should be fixed. It renders everything anyways, so can render all at once to get correct highlighting.

**Comment by wxiaoguang** [comment]
Fix code highlighting on blame page #36157
